lib/scrub/autoscrub.rb

```diff
@@ -30,12 +30,14 @@ def initialize(path)
         Logger.new(early_scrub_logger_path)
       end
       Services.scrub_logger.info("Getting basic info from #{@path}")
+      # Gotta fix encoding before we start looking in the file.
+      @encoding = Utils::Encoding.new(@path)
+      convert_to_utf8 unless @encoding.ascii_or_utf8?
```

Comment:
It was surprising to me to see this move to the constructor in so far as I'm not expecting `AutoScrub.new(...)` to potentially have side effects or to take a long time to complete, so I might consider moving this part back to `run`. On the other hand this only gets called from `ScrubRunner` and so perhaps it doesn't really matter - it's unlikely this is going to pop up somewhere else and cause confusion later on. Also IIRC some of the other `Scrub::` class constructors might have side effects insofar as opening output files, etc, anyway?

Reply:
I thought I was solving a problem that, it turns out, didn't exist. But I didn't, so yeah it's going back into `run`.